tools/ocp_version_update/update_default_release_versions_to_latest.py

```diff
@@ -333,50 +369,129 @@ def main(args):
+
+    if updates_made:
+        with open(os.path.join(ASSISTED_SERVICE_CLONE_DIR, DEFAULT_RELEASE_IMAGES_FILE), 'w') as outfile:
             json.dump(updated_version_json, outfile, indent=4)
-        verify_latest_config()
 
-        if dry_run:
-            return
+    return updates_made, updates_made_str
+
+
+def update_os_images_json(default_os_images_json, updates_made, updates_made_str, dry_run):
+    updated_version_json = copy.deepcopy(default_os_images_json)
+
+    for index, os_image in enumerate(default_os_images_json):
+        openshift_version = os_image["openshift_version"]
+        if openshift_version in SKIPPED_MAJOR_RELEASE:
+            logger.info(f"Skipping {openshift_version} listed in the skip list")
+            continue
+
+        cpu_architecture = os_image["cpu_architecture"]
+        if cpu_architecture != "x86_64":
+            logger.info(f"Skipping non-x86_64 OS images for now")
+            continue
+
+        rchos_image_url = os_image['url']       
+        rhcos_default_release = get_rchos_release_from_default_version_json(rchos_image_url, RHCOS_LIVE_ISO_REGEX)
+        rhcos_latest_of_releases = get_all_releases(RHCOS_RELEASES.format(minor=openshift_version))
+        rhcos_latest_release = get_latest_rchos_release_from_minor(openshift_version, rhcos_latest_of_releases)
+
+        if (not is_pre_release(rhcos_latest_release) and rhcos_default_release != rhcos_latest_release) or dry_run:
+            updates_made.add(openshift_version)
+            updates_made_str.add(f"rchos {rhcos_default_release} -> {rhcos_latest_release}")
+
+            logger.info(f"New latest rhcos release available, {rhcos_default_release} -> {rhcos_latest_release}")
+            updated_version_json[index]["url"] = updated_version_json[index]["url"].replace(rhcos_default_release, rhcos_latest_release)
+            updated_version_json[index]["rootfs_url"] = updated_version_json[index]["rootfs_url"].replace(rhcos_default_release, rhcos_latest_release)
+
+            if dry_run:
+                rhcos_version_from_iso = "8888888"
+            else:
+                rhcos_version_from_iso = get_rchos_version_from_iso(rhcos_latest_release, RCHOS_LIVE_ISO_URL.format(minor=openshift_version, version=rhcos_latest_release))            
+            
+            updated_version_json[index]["version"] = rhcos_version_from_iso
+
+    if updates_made:
+        with open(os.path.join(ASSISTED_SERVICE_CLONE_DIR, DEFAULT_OS_IMAGES_FILE), 'w') as outfile:
+            json.dump(updated_version_json, outfile, indent=4)
+
+    return updates_made, updates_made_str
+
+
+def create_jira_task(updates_made_str, dry_run, args):
+    logger.info(f"changes were made on the following versions: {updates_made_str}")
+
+    versions_str = ", ".join(updates_made_str)
+
+    if dry_run:
+        jira_client, task = None, "TEST-8888"
+    else:
+        jira_client, task = create_task(args, TICKET_DESCRIPTION + " " + versions_str)
```

Comment:
`jira_client` is never used.

```suggestion
        _, task = None, "TEST-8888"
    else:
        _, task = create_task(args, TICKET_DESCRIPTION + " " + versions_str)
```

Reply:
right, done